server/app.js

```diff
@@ -51,24 +51,25 @@ app.get("/courses/:courseId", (req, res) => {
 // Get All Students In A Course
 app.get("/courses/:courseId/students", (req, res) => {
   const { courseId } = req.params;
-  Student.find({ course: courseId }).then((students) => {
-    if (students.course !== courseId) {
+  Course.findById(courseId).then((course) => {
+    if (course) {
+      Student.find({ course: courseId }).then((students) => {
+        console.log(students);
+        res.json(students);
+        res.status(200);
+      });
```

Comment:
Looking at the endpoint `url` I think you only need this bit here. All students will be filtered based on the quey params, so you don't really need to look up any course